### PR TITLE
Aks node pool:  add node_taints && max_pods

### PR DESCRIPTION
**Pull request by romainFocone**
##### SUMMARY

Aks add node pool config node_taints && max_pods

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME
azure_rm_aks.py

### CODE REVIEW

File: plugins/modules/azure_rm_aks.py [hunk 1]
```diff
@@ -81,6 +81,11 @@
                 description:
                     - To enable auto-scaling.
                 type: bool
+                :param max_pods: Maximum number of pods that can run on a node.
```
Comment (haiyuazhang) [inline plugins/modules/azure_rm_aks.py]: this line will cause sanity check error, is it added by accident?

File: plugins/modules/azure_rm_aks.py [hunk 2]
```diff
@@ -111,6 +116,10 @@
                     - 1
                     - 2
                     - 3
+            node_taints:
+                description:
+                    - Taints added to new nodes during node pool create and scale
+                type: list
```
Comment (haiyuazhang) [inline plugins/modules/azure_rm_aks.py]: the sanity test shows there is a trailing empty space in the end of this line.

File: plugins/modules/azure_rm_aks.py (hunk 1 above)
Comment (Fred-sun) [inline plugins/modules/azure_rm_aks.py]:
```suggestion

```

File: plugins/modules/azure_rm_aks.py (hunk 2 above)
Comment (Fred-sun) [inline plugins/modules/azure_rm_aks.py]:
```suggestion
                type: list
```

**Comment by Fred-sun** [comment]
@romainFocone Thank you for submitting PR. I used the PR test you submitted and got the following errors. Did you test the newly added parameters, and how did it turn out?
```e.g.
"Error creating the AKS instance: Pods not in Running status: kubernetes-dashboard-64ccc85575-6lcfg,metrics-server-7f5b4f6d8c-b2f6v,tunnelfront-fd66db9b-h5sgs,coredns-79766dfd68-97n5s,coredns-autoscaler-66c578cddb-7f5jj"

**Comment by Fred-sun** [comment]
Kindly ping!

**Comment by romainFocone** [comment]
Hello, sorry for the latency, i think it's because the node pool with a taint block the other services to run if don't have taint.
In my case I have a script who add another node pool because it's currently impossible to do that directly with ansible-collection.
Normally if I add taint on the service that you have listed in test files, it should be ok, I will try.

**Comment by Fred-sun** [comment]
> Hello, sorry for the latency, i think it's because the node pool with a taint block the other services to run if don't have taint.
> In my case I have a script who add another node pool because it's currently impossible to do that directly with ansible-collection.
> Normally if I add taint on the service that you have listed in test files, it should be ok, I will try.

Any update for this?

**Comment by Fred-sun** [comment]
@romainFocone "max_pods" has been added by #440(has been merged), Please update your PR and remove this feature in your PR! Thank you very much!

**Pull request closed without merging**